### ISSUE TITLE
[BUG]: Error while updating 'SelectedTextTrack'

### ISSUE THREAD

**Issue by s43** [TheWidlarzGroup/react-native-video#3592]
### Version

v6 (Beta)

### What platforms are you having the problem on?

Android

### Architecture

Old architecture

### What happened?

When I try to change value of the prop selectedTextTrack, the app crashes and throws this error :  

![Screenshot_1711015021](https://github.com/react-native-video/react-native-video/assets/4326571/8ef67da6-97f2-4600-bdaa-7b16788b3548)

### Reproduction

repository link

### Reproduction

Step to reproduce this bug are: 
Simple implementation of the Video component with textTracks, and controlled selectedTextTrack prop.

**Comment by freeboub** [comment]
It looks like the issue is on the app side, can you detail how you select a track please (by lang, index, ... ) ?

**Comment by s43** [comment]
I am using the Index type, I've created a functional component that lists the retrieved text tracks with **onTextTracks** from an MKV source ( video's source ). and I do store the textTrack index, here's the video implementation 

```

<Video
      ref={ref}
      onTextTracks={handleTextTracks}
      onAudioTracks={handleAudioTracks}
      source={{
          uri: "https://github.com/ietf-wg-cellar/matroska-test-files/raw/master/test_files/test5.mkv",
      }}
      style={[
          {
              width,
              height,
          },
      ]}
      selectedTextTrack={
          metaData?.textTrack
              ? {
                    type: SelectedTrackType.INDEX,
                    value: metaData.textTrack,
                }
              : undefined
      }                    
      disableFocus
      resizeMode={ResizeMode.COVER}
      repeat={false}
  />
```

**Comment by freeboub** [comment]
did you check type of metaData.textTrack ? it should be number, not a text
notice that you will not be able to select index 0...

**Comment by s43** [comment]
Yes it is a number, I do store the same value as the "index" key within the textTracks received from onTextTracks

**Comment by freeboub** [comment]
I reproduce, let me check !

**Comment by s43** [comment]
Thank you, the problem only appears to happen when I use the index type, the subtitles gets updated with the language type tho.

**Comment by freeboub** [comment]
@s43 please see linked PR. You can test it now !

**Comment by s43** [comment]
That fixed the issue, Thank you Oliver!

**Comment by freeboub** [comment]
Will be available in 6.0.0
-beta.7
